wrappers/python/zxing.cpp

```diff
@@ -106,17 +150,61 @@ Results read_barcodes(py::object _image, const BarcodeFormats& formats, bool try
 	return read_barcodes_impl(_image, formats, try_rotate, try_downscale, text_mode, binarizer, is_pure, ean_add_on_symbol);
 }
 
-Image write_barcode(BarcodeFormat format, std::string text, int width, int height, int quiet_zone, int ec_level)
+class WriteResult
```

Comment:
The custom class is the only way I found to have a buffer and add custom attributes, like `__array_interface__` to it.

`__array_interface__` itself seems to be just a conversion helper which contains pretty much the same data as a buffer protocol type.

Reply:
Another thing to note here is that PIL (see `Image.fromarray`) never uses the `__array_interface__["data"]` key and instead always expects the object itself to be the buffer.

Reply:
Also, just using a dictionary won't work because you can't add attributes to it and `__array_interface__` is an attribute. Adding it as a key in the dictionary is not enough.

Reply:
I had a glace over the `__array_interface__` docs you linked but it left me bit puzzled. It seems to me that the numpy people consider the buffer protocol to be the state of the art way to share n-dimensional arrays of numbers between python extensions. And `__array_interface__` seems like the outdated/alternative option. If that would be right, then I would assume returning a 'buffer' would be the 'right' thing to do and adding the `__array_interface__` attribute might be more of a nice to have for 'older' code. Did you get the same impression? Am I missing something?

Reply:
Yes, that is my understanding as well. I added `__array_interface__` support to better support code that doesn't do buffers yet.

I still wonder whether we could return `WriteResult` or `py::buffer` from `write_barcode` directly so we have better typing here.

Reply:
I think a found a somewhat cleaner solution for the `WriteResult` struct. I'll add it to this PR later.

Reply:
done